pkg/generators/common_blocks/pod/externalsecrets_test.go

```diff
@@ -0,0 +1,277 @@
+	"github.com/go-test/deep"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func TestGenerateExternalSecretFn(t *testing.T) {
+	type args struct {
+		name            string
+		namespace       string
+		secretStoreName string
+		secretStoreKind string
+		refreshInterval metav1.Duration
+		labels          map[string]string
+		opts            interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want *externalsecretsv1alpha1.ExternalSecret
+	}{
+		{
+			name: "Generates a new ExternalSecret from an Options struct",
+			args: args{
+				name:            "my-secret",
+				namespace:       "test",
+				secretStoreName: "vault-mgmt",
+				secretStoreKind: "ClusterSecretStore",
+				refreshInterval: metav1.Duration{Duration: 60 * time.Second},
+				labels:          map[string]string{},
+				opts: struct {
+					Option1 EnvVarValue `env:"OPTION1"`
+					Option2 EnvVarValue `env:"OPTION2" secret:"my-secret"`
+					Option3 EnvVarValue `env:"OPTION3" secret:"my-secret"`
+					Option4 EnvVarValue `env:"OPTION4" secret:"other-secret"`
+					Option5 EnvVarValue `env:"OPTION5" secret:"not-set"`
+				}{
+					Option1: &ClearTextValue{Value: "value1"},
+					Option2: &SecretValue{Value: saasv1alpha1.SecretReference{
+						FromVault: &saasv1alpha1.VaultSecretReference{Key: "key2", Path: "path2"}}},
+					Option3: &SecretValue{Value: saasv1alpha1.SecretReference{
+						FromVault: &saasv1alpha1.VaultSecretReference{Key: "key3", Path: "path3"}}},
+					Option4: &SecretValue{Value: saasv1alpha1.SecretReference{
+						FromVault: &saasv1alpha1.VaultSecretReference{Key: "key3", Path: "path3"}}},
+				},
+			},
+			want: &externalsecretsv1alpha1.ExternalSecret{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       externalsecretsv1alpha1.ExtSecretKind,
+					APIVersion: externalsecretsv1alpha1.ExtSecretGroupVersionKind.GroupVersion().String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-secret",
+					Namespace: "test",
+					Labels:    map[string]string{},
+				},
+				Spec: externalsecretsv1alpha1.ExternalSecretSpec{
+					SecretStoreRef: externalsecretsv1alpha1.SecretStoreRef{
+						Name: "vault-mgmt",
+						Kind: "ClusterSecretStore",
+					},
+					Target:          externalsecretsv1alpha1.ExternalSecretTarget{Name: "my-secret"},
+					RefreshInterval: &metav1.Duration{Duration: 60 * time.Second},
+					Data: []externalsecretsv1alpha1.ExternalSecretData{
+						{
+							SecretKey: "OPTION2",
+							RemoteRef: externalsecretsv1alpha1.ExternalSecretDataRemoteRef{
+								Key:      "path2",
+								Property: "key2",
+							},
+						},
+						{
+							SecretKey: "OPTION3",
+							RemoteRef: externalsecretsv1alpha1.ExternalSecretDataRemoteRef{
+								Key:      "path3",
+								Property: "key3",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Generates other ExternalSecret from the same Options struct (see previous test)",
```

Comment:
Adding a new test case, using refreshInterval not set to 60s (the default value) or event parsing a string to duration. 

And other converting a string, like `2m` to duration and then validating it?

Reply:
That's true, I forgot about it, I have added to the ExternalSecret resource unit test https://github.com/3scale-ops/saas-operator/pull/150/commits/b4d08fdf75fe3c04aff4e421969e500f9c22d97c:
- from `120 second` minute to `120 second`
- from `2 minute` to `120 second`


The default value (`60s`) will be used when having the resource inside a workload controller (not used yet), which I will cover the possibility to override the default `60s` values with any other value via `controller.spec.secretName.fromVault.refreshInterval`

Reply:
Awesome.